pkg/reconciler/route/route.go

```diff
@@ -272,7 +287,7 @@ func (c *Reconciler) tls(ctx context.Context, host string, r *v1alpha1.Route, tr
 			}
 			// TODO: we should only mark https for the public visible targets when
 			// we are able to configure visibility per target.
-			setTargetsScheme(&r.Status, cert.Spec.DNSNames, "https")
+			setTargetsScheme(&r.Status, allDomains, "https")
```

Comment:
I don't think we should use `allDomains` here and below as it will mark all targets to https.
We should use `dnsNames.List()`.